test/integration/tests/nv.sh

```diff
@@ -64,11 +64,11 @@ cleanup "no-shut-down"
 tpm2_clear
 
 #Test default values for the hierarchy "-a" parameter
-tpm2_nvdefine -Q -x $nv_test_index -s 32 -t "ownerread|policywrite|ownerwrite"
+tpm2_nvdefine -Q -x $nv_test_index -s 32 -A "ownerread|policywrite|ownerwrite"
```

Comment:
we need to get this as a non-capital argument, is "-a" used?

Here is a list of the current ones:
```
$ grep -rn "case '.'" | cut -d"'" -f2-2 | sort | uniq
?
a
A
c
C
D
e
E
f
F
g
G
h
H
i
k
K
l
L
m
n
N
o
O
p
P
q
Q
r
s
S
t
T
u
U
v
V
x
X
Z
```
I could consider -y for attributes, but I was hoping one day to be able to specify the public portion of an object via YAML. So I wouldn't want to use that.

Perhaps -w "like with these attributes" thoughts?

Reply:
Looks like you want to swap -o for -w in the next patch, so maybe -b

Reply:
`-a` is mostly used for specifying hierarchy :

```
>$ grep -R "'a'[[:space:]]*}" | sed "s%[[:space:]]*%%g"

tools/tpm2_startauthsession.c:{"auth-policy-session",no_argument,NULL,'a'},
tools/tpm2_createpolicy.c:{"auth-policy-session",no_argument,NULL,'a'},
tools/tpm2_nvread.c:{"hierarchy",required_argument,NULL,'a'},
tools/tpm2_evictcontrol.c:{"hierarchy",required_argument,NULL,'a'},
tools/tpm2_nvincrement.c:{"hierarchy",required_argument,NULL,'a'},
tools/tpm2_nvwrite.c:{"hierarchy",required_argument,NULL,'a'},
tools/tpm2_hash.c:{"hierarchy",required_argument,NULL,'a'},
tools/tpm2_createprimary.c:{"hierarchy",required_argument,NULL,'a'},
tools/tpm2_loadexternal.c:{"hierarchy",required_argument,NULL,'a'},
tools/tpm2_nvreadlock.c:{"hierarchy",required_argument,NULL,'a'},
tools/tpm2_nvdefine.c:{"hierarchy",required_argument,NULL,'a'},
tools/tpm2_nvrelease.c:{"hierarchy",required_argument,NULL,'a'},
```

My gears are currently grinded on auth-policy-session (`-p` instead ?).

My goal is to sanctuarize key options : `-o` for output, `-i` for input, etc...

I don't have any objections on swapping `-A` to `-b`. I actually missed the "capitalized is parent" policy on that one

Reply:
> My gears are currently grinded on auth-policy-session (-p instead ?).
I'd like to keep -p|-P for authvalue consistent across all tools. For tpm2_startauthsession, -a is for policy over trial type of session. perhaps just long option it --policy and even add --trial? I'd say the same approach for tpm2_createpolicy.

ETOOMANYOPTIONS this gets complicated/monotonous quickly.

Reply:
The current behaviour on whether if we start a policy or a trial session rely only on the "-a" toggle.

For the sake of simplicity, it would be easier to state that default behaviour is `SE_TRIAL` and the `--policy` allow a `SE_POLICY` session